c/extras/aggregator.cc

```diff
@@ -66,13 +44,37 @@ Aggregator::Aggregator(int32_t n_bins_in, int32_t nx_bins_in, int32_t ny_bins_in
 }
 
 
-void Aggregator::aggregate(DataTable* dt_exemplars, DataTable* dt_members) {
-  switch (dt_exemplars->ncols) {
-    case 1:  group_1d(dt_exemplars, dt_members); break;
-    case 2:  group_2d(dt_exemplars, dt_members); break;
-    default: group_nd(dt_exemplars, dt_members);
+DataTable* Aggregator::aggregate(DataTable* dt) {
+  DataTable* dt_members = nullptr;
+  DataTable* dt_double = nullptr;
+  Column** cols_double = dt::amalloc<Column*>(dt->ncols + 1);
+  Column** cols_members = dt::amalloc<Column*>(static_cast<int64_t>(2));
+
+  for (int32_t i = 0; i < dt->ncols; ++i) {
+    LType ltype = info(dt->columns[i]->stype()).ltype();
+    switch (ltype) {
+      case LType::BOOL:
+      case LType::INT:
+      case LType::REAL: cols_double[i] = dt->columns[i]->cast(SType::FLOAT64); break;
+      default:          cols_double[i] = dt->columns[i]->shallowcopy();
+    }
+  }
+
+  cols_double[dt->ncols] = nullptr;
+  cols_members[0] = Column::new_data_column(SType::INT32, dt->nrows);
+  cols_members[1] = nullptr;
+  dt_double = new DataTable(cols_double);
+  dt_members = new DataTable(cols_members);
```

Comment:
Here, on the other hand, it would make perfect sense to have `dt_double` and `dt_members` as `std::unique_ptr`s. Because currently if any of the `group_*` functions fails, the constructed `DataTable`s will leak.

The `dt_members` can then be returned as `std::unique_ptr` as well.

Reply:
Right, I guess this is a reason why I kept `dt_members` as `DataTablePtr` in `PyObject* aggregate(PyObject*, PyObject* args)`. Was not sure what is the best place to create the required tables and converged somewhere in the middle. Fixing it now.